dynamo/plot/scVectorField.py

```diff
@@ -498,33 +704,46 @@ def cell_wise_vectors(
         sort=sort,
         save_show_or_return="return",
         frontier=frontier,
+        projection=projection,
         **s_kwargs_dict,
         return_all=True,
     )
 
-    if type(axes_list) == list:
-        for i in range(len(axes_list)):
-            axes_list[i].quiver(
-                df.iloc[:, 0],
-                df.iloc[:, 1],
-                df.iloc[:, 2],
-                df.iloc[:, 3],
+    # single axis output
+    if type(axes_list) != list:
+        axes_list = [axes_list]
+    x0, x1 = df.iloc[:, 0], df.iloc[:, 1]
+    v0, v1 = df.iloc[:, 2], df.iloc[:, 3]
+
+    if projection == "3d":
+        x0, x1, x2 = df.iloc[:, 0], df.iloc[:, 1], df.iloc[:, 2]
+        v0, v1, v2 = df.iloc[:, 3], df.iloc[:, 4], df.iloc[:, 5]
+
+    for i in range(len(axes_list)):
+        ax = axes_list[i]
+        if projection == "2d":
+            ax.quiver(
```

Comment:
here obviously you need to use the x0/x1/x2 and v0/v1/v2 from the corresponding genes. Looks like you use the same X and V for all axes? this is wrong

Reply:
This projection 2d code is equivalent to the old code version for showing quiver plot X,V matrices selected by method code above. v0 v1 v2 are velocity embeddings. The behavior of this method will be changed if I do so and we have not discussed if this is allowed because it can cause extra problems on old analysis notebooks.

Reply:
Regardless of equivalence issue, I double checked the code in the old version. What it does is to keep X, V the same and show quivers in each color ax subplot. The coloring of points are from genes. Do we have 3d more higher dimension velocity vectors for each gene at each point? I do not think so and we can discuss it.

Reply:
are you sure it is the same? note that I pointed out all axes used the same x0-x2 and v0-v2 which is wrong